box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestDownload.java

```diff
@@ -203,60 +216,81 @@ public BoxDownload onResponse(Class clazz, BoxHttpResponse response) throws Ille
             String contentType = response.getContentType();
             long contentLength = -1;
 
-            if (response.getResponseCode() == HttpStatus.SC_OK || response.getResponseCode() == HttpStatus.SC_PARTIAL_CONTENT){
+            if (response.getResponseCode() == HttpStatus.SC_ACCEPTED) {
+                try {
+                    if (mNumRetries < DEFAULT_NUM_RETRIES) {
+                        mNumRetries++;
+                        // First try with retry-after value, then use exponential back-off
+                        String value = response.getHttpURLConnection().getHeaderField("Retry-After");
+                        if (!SdkUtils.isBlank(value)) {
+                            int retryAfterSeconds = Integer.parseInt(value);
+                            retryAfterSeconds = retryAfterSeconds > 0 ? retryAfterSeconds : 1;
+                            mRetryAfterMillis = retryAfterSeconds * 1000;
+                        }
```

Comment:
You might be right if there is always a header it might not get into the doubling statement.
I will verify with Brian later.